plugins/proxy-backend/src/service/router.ts

```diff
@@ -104,15 +99,18 @@ export function buildMiddleware(
     ].map(h => h.toLocaleLowerCase()),
   );
 
-  // only forward the allowed headers in client->backend
-  fullConfig.onProxyReq = (proxyReq: http.ClientRequest) => {
```

Comment:
You need to update the tests in [router.test.ts](https://github.com/backstage/backstage/blob/9d8515ba68470bd0eb0ee1b654156857e8bd52d7/plugins/proxy-backend/src/service/router.test.ts#L136) as they explicitly check if the `onProxyReq` is setup correctly and they now need to be updated to check if the `filter` function is setup correctly.

Reply:
Thanks, am fixing the tests now. I think I'll need to separate the header processing into a different function that's called from the filter so I can test it.

Reply:
Hi - I didn't separate out the filtering in the end, but I've updated the tests to check that the correct filtering is applied to `req.headers` after `filter()` is called